### DISCUSSION TITLE
Negative maximum length in tokens truncates from the left, suffix becomes prefix

### DISCUSSION THREAD

**Discussion by haphi** [polybar/polybar#2877] [Ideas]
I think it would be useful to be able to truncate tokens from the front instead of the back by using a negative value for the maximum length:

```
label = "%token:0:-8"
```
would result in a longer string, i.e. `asdfghjkl` to be truncated to `sdfghjkl` instead of `asdfghjk`.

The truncation suffix would instead be prepended as prefix.

This relates to #2789, and I feel it is a natural extension - the sign of the minimum width determines whether to add padding left or right, whereas the sign of the maximum width would determine whether to truncate from the left or the right.

This is especially useful when displaying file or directory paths, as the most important information will be in the back of the string - in any case that was my motivation to work on this.

I implemented this on a branch that I would be happy to polish for a MR if wanted.

**Comment by patrick96** [comment]
Hmm, yeah, prefix truncation would actually be a good idea. I'd be happy to review the PR :)
